libphx/src/file.rs

```diff
@@ -12,15 +11,19 @@ pub struct File {
 
 #[no_mangle]
 pub extern "C" fn File_Exists(path: *const libc::c_char) -> bool {
-    match fs::metadata(ffi::PtrAsSlice(path)) {
+    file_exists(&path.convert())
```

Comment:
I left only `convert() -> &str` variant with your implementation from `PtrAsSlice` and where it's needed I use `.to_string()`.

Reply:
I think that's definitely an improvement!

A couple of other thoughts:
* `convert()` feels a little unclear to me what it's doing, it doesn't do the same as `into()` for example (which is more obvious). Most of the Rust standard library uses `to_xx()` methods to describe conversions. I think it would be nicer if this was `to_slice()` or `to_str()`, and the conversion the other way around is `to_ptr()`
* `convert().to_string()` is fine but also feels a little verbose to write, it might be convenient to have a `.to_string()` that wraps that 😄

Reply:
Let me see what I can do :smiley:

Reply:
Just added ConvertToOwned::to_owned_value helper to replace .convert().to_string()